### ISSUE TITLE
[Function _n] What wrong?

### ISSUE THREAD

**Issue by ttn10** [Otto42/theme-check#175]
This code of Woocommerce Plugin:

`echo apply_filters( 'woocommerce_available_download_count', '<span class="woocommerce-Count count">' . sprintf( _n( '%s download remaining', '%s downloads remaining', $download['downloads_remaining'], 'domain' ), $download['downloads_remaining'] ) . '</span> ', $download );`

but Them check is warning:

`WARNING: Found a translation function that has an incorrect number of arguments. Function _n, with the arguments '%s download remaining', '%s downloads remaining', downloads_remaining, 'domain'`

I see code here:
https://developer.wordpress.org/reference/functions/_n/
// What wrong?

**Comment by grappler** [comment]
At the first glance I can't see an issue with with code with `_n`.

As for the text domain. The text domain must the same as the theme name. So if the theme is called. Olympic Champion then the text domain would be `olympic-champion`.

**Comment by Otto42** [comment]
The check itself is not perfect. It's seeing the string in  $download['downloads_remaining'] and counting that as a possible argument which it is not expecting.

There's nothing wrong with your code. If you want to avoid the false error as a workaround for now, use a temporary variable to hold  $download['downloads_remaining'] and use that in the _n call instead.

**Comment by jrfnl** [comment]
Looking at this check, a lot of this should probably be removed in favour of the WPCS I18n sniff which already handles this correctly.

**Comment by joyously** [comment]
But the sniff is a totally different context.

**Comment by jrfnl** [comment]
@joyously What do you mean ? Isn't the TRTCS sniffer supposed to be (or already) integrated in the Theme Check plugin ?

**Comment by joyously** [comment]
No, they are totally separate.

**Comment by dingo-d** [comment]
Nope, we're pausing the sniffer development because I just don't have the time to work on it (especially since it'S something I'm doing it on my free time).

**Comment by pattonwebz** [comment]
I will make the time to take oversee the sniffer projects :)
